numba/cuda/tests/cudapy/test_sync.py

```diff
@@ -71,6 +90,30 @@ def test_dyn_shared_memory(self):
         compiled[1, shape, 0, ary.size * 4](ary)
         self.assertTrue(np.all(ary == 2 * np.arange(ary.size, dtype=np.int32)))
 
+    def test_threadfence_codegen(self):
+        compiled = cuda.jit("void(int32[:])")(use_threadfence)
+        ary = np.zeros(10, dtype=np.int32)
+        compiled[1, 1](ary)
```

Comment:
Actually, as I understand it, the synchronization model of thread fences would not make the test work with several threads even in CUDA, as that would require atomic increments. I wonder if there's any simple way to test the functionality...